tokio/src/runtime/thread_pool/mod.rs

```diff
@@ -51,10 +51,17 @@ impl ThreadPool {
         handle_inner: HandleInner,
         before_park: Option<Callback>,
         after_unpark: Option<Callback>,
+        before_task_poll: Option<Callback>,
     ) -> (ThreadPool, Launch) {
         let parker = Parker::new(driver);
-        let (shared, launch) =
-            worker::create(size, parker, handle_inner, before_park, after_unpark);
+        let (shared, launch) = worker::create(
+            size,
+            parker,
+            handle_inner,
+            before_park,
```

Comment:
This can be a follow up, but it seems time to create a `Callbacks` structure to pass around :)

Reply:
Also, not necessarily for you to do, if anyone else see this and is motivated, they can jump on ^^

Reply:
Do I understand this correctly: `ThreadPool::new` should have a `Callbacks` struct as 4th parameter that contains all `Option<Callback>` functions?